assets/app/scripts/controllers/create/createFromImage.js

```diff
@@ -64,10 +66,18 @@ angular.module("openshiftConsole")
           var imageName = scope.imageTag;
           DataService.get("imagestreamtags", imageStream.metadata.name + ":" + imageName, {namespace: scope.namespace}).then(function(imageStreamTag){
               scope.image = imageStreamTag.image;
-              angular.forEach(imageStreamTag.image.dockerImageMetadata.ContainerConfig.Env, function(entry){
+              var env = $parse('dockerImageMetadata.ContainerConfig.Env')(imageStreamTag.image) || [];
+              angular.forEach(env, function(entry){
                 var pair = entry.split("=");
                 scope.deploymentConfig.envVars[pair[0]] = pair[1];
               });
+
+              scope.routing.portOptions = ApplicationGenerator.parsePorts(imageStreamTag.image);
+              if (scope.routing.portOptions.length){
+                scope.routing.targetPort = scope.routing.portOptions[0];
```

Comment:
I see that's the default. Would it be better to remove the default `include:true` and set to true here instead of reverting to false in the negative lookup case?